Cabal/Distribution/Simple/SrcDist.hs

```diff
@@ -149,55 +162,64 @@ listPackageSourcesOrdinary :: Verbosity
                            -> [PPSuffixHandler]
                            -> IO [FilePath]
 listPackageSourcesOrdinary verbosity pkg_descr pps =
+-- TODO: We should remove the verbosity parameter.
+--       Function defaultPackageDesc isn't using it internally.
   fmap concat . sequenceA $
   [
     -- Library sources.
     fmap concat
-    . withAllLib $ \Library { exposedModules = modules, libBuildInfo = libBi } ->
-     allSourcesBuildInfo libBi pps modules
+    . withAllLib $ \Library {exposedModules = ex, libBuildInfo = bi} -> do
+       let exposed = filterAutogenModules pkg_descr bi ex
+       let filteredBi = filterAutogenBuildInfo pkg_descr bi
+       moduleSrcs <- findModulesFiles (hsSourceDirs bi) pps
+                                      (exposed ++ otherModules filteredBi)
+       return $ moduleSrcs ++ cSources filteredBi ++ jsSources filteredBi
```

Comment:
The `cSources filteredBi ++ jsSources filteredBi` code appears five times but is used on different `BuildInfo` structures. Five places seems to have the same 4-line code but there are three distinct cases that can't be extracted on a single functions without being unreadable.